src/api/localStorage.js

```diff
@@ -0,0 +1,110 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+    console.log('User Added: ', user);
+    return this.users;
+  } 
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+    return this.users;
+  }
+ 
+  getUser(id) {
+    const user = this.findUserMatch(id);
+    console.log('getUser() -> User Found: ', user);
+    return user;
+  }
+
+  getAllUsers() {
+    console.log('getAllUsers() -> Users Found: ', this.users);
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    const editObj = {};
+    editObj[`${editName}`] = `${editValue}`;
+    const updatedUser = Object.assign(this.getUser(id), editObj);
+    console.log('Updated User: ', updatedUser);
+    this.createUser(updatedUser);
+  }
+
+  replaceUser(id, update) {
+    const index = this.getUserIndex(id);
+    const replacedUser = Object.assign(
+      this.users[index], update);
+    console.log('replaceUser() -> Replaced User: ', replacedUser)
+    return replacedUser;
+  }
+
+  deleteUser(id) {
+    this.updateUsers();
+    this.users.forEach((user, index) => {
+      if (user.id.toString() === id.toString()) {
```

Comment:
Super nitpick but you _could_ also use 

```
String(user.id) === String(id)
```

Reply:
Are the `id` properties not already stored as strings?

Reply:
Not at the moment. I have the generateId() running from createUser() to generate a random id each time a user is created. Was creating it in the component props initially but was having an issue with the same id being used for every user.